classes/services/SubmissionService.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @file classes/services/SubmissionService.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionService
+ * @ingroup services
+ *
+ * @brief Extends the base submission helper service class with app-specific
+ *  requirements.
+ */
+
+namespace App\Services;
+
+class SubmissionService extends PKPSubmissionService {
+
+	/**
+	 * Initialize hooks for extending PKPSubmissionService
+	 */
+    public function __construct() {
+		parent::__construct();
+
+		\HookRegistry::register('Submission::getSubmissionList::queryBuilder', array($this, 'getSubmissionListQueryBuilder'));
+		\HookRegistry::register('Submission::listQueryBuilder::get', array($this, 'getSubmissionListQueryObject'));
+		\HookRegistry::register('Submission::toArray::defaultParams', array($this, 'toArrayDefaultParams'));
+		\HookRegistry::register('Submission::toArray::output', array($this, 'toArrayOutput'));
+	}
+
+	/**
+	 * Run app-specific query builder methods for getSubmissionList
+	 *
+	 * @param $hookName string
+	 * @param $args array [
+	 *		@option QueryBuilders\SubmissionListQueryBuilder $submissionListQB
+	 *		@option int $contextId
+	 *		@option array $args
+	 * ]
+	 *
+	 * @return QueryBuilders\SubmissionListQueryBuilder
+	 */
+	public function getSubmissionListQueryBuilder($hookName, $args) {
+		$submissionListQB =& $args[0];
+		$contextId = $args[1];
+		$args = $args[2];
+
+		if (!empty($args['sectionIds'])) {
+			$submissionListQB->filterBySections($args['sectionIds']);
+		}
+
+		return $submissionListQB;
+	}
+
+	/**
+	 * Add app-specific query statements to the list get query
+	 *
+	 * @param $hookName string
+	 * @param $args array [
+	 *		@option object $queryObject
+	 *		@option QueryBuilders\SubmissionListQueryBuilder $queryBuilder
+	 * ]
+	 *
+	 * @return object
+	 */
+	public function getSubmissionListQueryObject($hookName, $args) {
+		$queryObject =& $args[0];
+		$queryBuilder = $args[1];
+
+		$queryObject = $queryBuilder->appGet($queryObject);
```

Comment:
`appGet` doesn't appear to be used anywhere in the PKP library. Suggest removing its `abstract` definition there, and refactoring away from hooks here if a simple subclass extension is an option. Hooks are spaghetti.